dirty_cat/datasets/_ken_embeddings.py

```diff
@@ -0,0 +1,97 @@
+"""
+Get the Wikipedia embeddings for feature augmentation.
+"""
+import pandas as pd
+from sklearn.decomposition import PCA
+
+from dirty_cat.datasets import fetch_figshare
+
+
+def get_ken_embeddings(
+    types,
+    exclude=None,
+    pca_components=None,
+    emb_id="39142985",
+    emb_type_id="39266300",
+    suffix="",
+):
+    """Extract Wikipedia embeddings by type.
+
+    More details on the embeddings can be found on
+    https://soda-inria.github.io/ken_embeddings/.
+
+    Parameters
+    ----------
+    types : str
+        Types of embeddings to include. Write in lowercase.
+    exclude : str, default=None
+        Type of embeddings to exclude from the types search.
+    pca_components : int, default=None
+        Size of the dimensional space on which the embeddings will be projected
+        by a principal component analysis.
+        If None, the default dimension (200) of the embeddins will be kept.
+    emb_id : str
+        Figshare ID of the file containing all embeddings.
```

Comment:
It's a horrible API for the user to have to specify magic numbers like this.

We should have a correspondence table between user-comprehensible names (eg "ken_embedding") and ids. We could accept both strings, in which case they are converted to ids using this table, and numbers.

The challenges are now:
- Where does this table get stored. If it gets stored in dirty-cat, it is difficult to update: when we add a new release of ken_embedding we need to do a new release of dirty-cat (ouch). We might want to store it online and download it. The logic would then be that it is cached, but if there is a string that we do not recognize, we download a new version (just in case this look up table has been updated) before trying again to resolve this string, and if it doesn't work, only then do we raise an exception
- How to we make sure that the user can easily see which are the valid versions. An URL to the table stored in a human-readable format (CSV?) could provide an answer.

Reply:
I will start making the correspondence table.

Is maybe the best solution is to add this correspondence table to https://github.com/dirty-cat/datasets?

I think if we fetch it from there with the link e.g. https://raw.githubusercontent.com/dirty-cat/datasets/master/data/correspondence.csv , it will not change when we update the csv file and it will be update if re-downloaded?

Also, the user would be able to easily access it as we provide a link to it and the table will be visible directly due to a small number of rows.

WDYT?

Reply:
Sounds good!

Reply:
Ok it's ready! https://github.com/dirty-cat/datasets/pull/6